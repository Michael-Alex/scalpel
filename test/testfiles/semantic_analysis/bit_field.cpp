struct s
{
	int i: 5;
	mutable bool j: 3;
};

