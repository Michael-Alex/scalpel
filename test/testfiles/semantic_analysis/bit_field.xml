<namespace id="n0">
 <class name="s" id="c0">
  <destructor access="public" defined="false">
  </destructor>
  <bit_field name="i" size="0" access="public">
   <type>
    <fundamental_type type="int"/>
   </type>
  </bit_field>
  <bit_field name="j" size="0" access="public" mutable="true">
   <type>
    <fundamental_type type="bool"/>
   </type>
  </bit_field>
 </class>
</namespace>
