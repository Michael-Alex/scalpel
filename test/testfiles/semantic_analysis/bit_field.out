{
  "id": 0,
  "classes":
  [
    {
      "id": 0,
      "name": "s",
      "destructor":
      {
        "access": "public"
      },
      "bit fields":
      [
        {
          "id": 0,
          "name": "i",
          "size": 0,
          "access": "public",
          "mutable": "true",
          "type":
          {
            "fundamental type": "bool"
          }
        },
        {
          "id": 1,
          "size": 0,
          "access": "public",
          "type":
          {
            "fundamental type": "int"
          }
        },
        {
          "id": 2,
          "size": 0,
          "access": "public",
          "type":
          {
            "fundamental type": "unsigned int"
          }
        }
      ]
    }
  ]
}
